src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -192,17 +192,50 @@ export abstract class InstructorSessionBasePageComponent {
   }
 
   /**
-   * Copies the feedback session.
+   * Combines a {@link SessionsTableRowModel} and {@link CopySessionResult} to submit the copy session requests.
    */
-  copySession(model: SessionsTableRowModel, result: CopySessionResult): void {
-    this.copyFeedbackSession(model.feedbackSession, result.newFeedbackSessionName, result.copyToCourseId)
-        .subscribe((createdSession: FeedbackSession) => {
-          this.navigationService.navigateWithSuccessMessage(
-              this.router,
-              '/web/instructor/sessions/edit',
-              'The feedback session has been copied. Please modify settings/questions as necessary.',
-              { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
-        }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
+  copySessionTransformer(model: SessionsTableRowModel, result: CopySessionResult): FeedbackSession[] {
+    const copySessionRequests: Observable<FeedbackSession>[] = [];
+    result.copyToCourseList.forEach((copyToCourseId: string) => {
+      copySessionRequests.push(
+          this.copyFeedbackSession(model.feedbackSession, result.newFeedbackSessionName, copyToCourseId));
+    });
+    return this.copySession(copySessionRequests);
+  }
+
+  /**
+   * Submits the copy session requests.
+   */
+  copySession(copySessionRequests: Observable<FeedbackSession>[]): FeedbackSession[] {
+    const successMessage: string =
+        'The feedback session has been copied. Please modify settings/questions as necessary.';
+    const sessionList: FeedbackSession[] = [];
+    if (copySessionRequests.length === 1) {
+      copySessionRequests[0].subscribe((createdSession: FeedbackSession) => {
+        sessionList.push(createdSession);
+        this.navigationService.navigateWithSuccessMessage(this.router,
+            '/web/instructor/sessions/edit', successMessage,
+            { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
+      }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
+    } else if (copySessionRequests.length > 1) {
+      let isAnyFailed: boolean = false;
+      forkJoin(copySessionRequests).pipe(
+          tap((sessions: FeedbackSession[]) => {
+            sessionList.concat(sessions);
+          }),
+          catchError((error: any) => {
+            isAnyFailed = true;
+            return of(error);
+          }),
+      ).subscribe(() => {
+        if (isAnyFailed) {
```

Comment:
How can I retrieve the `courseId` from the request array?

Reply:
Take a look at how the submission page does it.

Reply:
The submission page has the luxury of having the request array creation and `forkJoin` submission happen within the same method scope, so the `catchError` block is constructed as part of the request. For me to do something similar here would require extensive code refactoring so this might take some time

Reply:
You can have `this.copyFeedbackSession` to construct the error message and just display here. There is no need to know which one is failing here.